From 2ed70ae3224858a567bdbb9cb3cd41a8b7f5318b Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:11:39 -0400
Subject: [alsa] Add SNDRV_PCM_STATE_PAUSED case in wait_for_avail function

Message-id: <140972823573-2-patchset-jkysela@redhat.com>
Patchwork-id: 90702
O-Subject: [RHEL 7.1 PATCH 001/442] ALSA: Add SNDRV_PCM_STATE_PAUSED case in wait_for_avail function
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

When the process is sleeping at the SNDRV_PCM_STATE_PAUSED
state from the wait_for_avail function, the sleep process will be woken by
timeout(10 seconds). Even if the sleep process wake up by timeout, by this
patch, the process will continue with sleep and wait for the other state.

Signed-off-by: JongHo Kim <furmuwon@gmail.com>
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: ed697e1aaf7237b1a62af39f64463b05c262808d)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/core/pcm_lib.c b/sound/core/pcm_lib.c
index 6e03b46..a210467 100644
--- a/sound/core/pcm_lib.c
+++ b/sound/core/pcm_lib.c
@@ -1937,6 +1937,8 @@ static int wait_for_avail(struct snd_pcm_substream *substream,
   case SNDRV_PCM_STATE_DISCONNECTED:
    err = -EBADFD;
    goto _endloop;
+  case SNDRV_PCM_STATE_PAUSED:
+   continue;
   }
   if (!tout) {
    snd_printd("%s write error (DMA or IRQ trouble?)\n",
-- 
1.7.1